ingestion/src/metadata/profiler/interface/pandas/pandas_profiler_interface.py

```diff
@@ -124,25 +118,12 @@ def _(
         import pandas as pd  # pylint: disable=import-outside-toplevel
 
         try:
-            row = []
+            row_dict = {}
             for metric in metrics:
-                row.append(
-                    metric().df_fn(
-                        [
-                            df.astype(object).where(pd.notnull(df), None)
-                            for df in self.dfs
-                        ]
-                    )
+                row_dict[metric.name()] = metric().df_fn(
+                    [df.where(pd.notnull(df), None) for df in self.dfs]
```

Comment:
Shouldn't we do that once before looping? if we have `n` metrics we'll be doing the same operation n times from what I understand.